### DISCUSSION TITLE
should if x is Type alter inferred type

### DISCUSSION THREAD

**Discussion by SiyahaS** [godotengine/godot-proposals#9922] [Scripting]
I have recently started working with Godot and while trying features I have noticed if I enable unsafe method/property access I get a lot of warnings which should not occur. 

```gdscript
var x := myscene.instantiate()
if x is Node2D:
    x.global_position = Vector2.ZERO # x is still inferred as Node instead of Node2D here
```
    
I think this kind of automatic type inference change would increase developer experience.

**Comment by AThousandShips** [comment]
Thank you for your proposal, consolidating in:
* https://github.com/godotengine/godot-proposals/issues/8530
